feat(playerMove.cs, spawner.cs)

GaMaius opened the pull request:
# 변경점 👍
 플레이어 움직임, 적 생성 및 플레이어 추적 스크립트 작성 및 적용

GaMaius committed:
feat(playerMove.cs, spawner.cs)

플레이어 움직임, 적 생성 및 플레이어 추적 스크립트 작성 및 적용

GaMaius committed:
feat(playerMove.cs)

플레이어 움직임에 따른 X축 플립 설정.

hwanginseop commented on Assets/Script/playerMove.cs:
```diff
@@ -0,0 +1,56 @@
+    }
+
+    // 그래픽 & 인풋 업데이트.
+    void Update()
```
안쓰는 Update는 비어있어도 다 잡히기때문에 지우는게 좋습니다~

hwanginseop commented on Assets/Script/playerMove.cs:
```diff
@@ -0,0 +1,56 @@
+    // 변수 선언.
+    public float moveSpeed = 1f;
+
+    Rigidbody2D rigid;
```
해당 변수는 사용되지 않으므로 지우는게 좋아보입니다.

hwanginseop commented on Assets/Script/playerMove.cs:
```diff
@@ -0,0 +1,56 @@
+public class playerMove : MonoBehaviour
+{
+    // 변수 선언.
+    public float moveSpeed = 1f;
```
Speed의 값을 매직넘버로 정해놓고 들어가는거 매우 좋은것 같습니다.

hwanginseop commented on Assets/Script/playerMove.cs:
```diff
@@ -0,0 +1,56 @@
+
+        if (h < 0)
+        {
+            render.flipX = true;
```
어느정도 기초적인 애니메이션의 구현을 시도한 점 매우 좋네요

hwanginseop commented on Assets/Script/spawner.cs:
```diff
@@ -0,0 +1,79 @@
+        curTime += Time.deltaTime;
+        if (curTime >= 2)
+        {
+            GameObject _clone = MakeRandomEnemy();
```
위에서 한 번 선언하였기 때문에 다시 선언할 필요는 없어보입니다.

hwanginseop commented on Assets/Script/spawner.cs:
```diff
@@ -0,0 +1,79 @@
+    public float speed = 1f;
+    float curTime = 0;
+
+    void Start()
```
위의 Update와 마찬가지로 지워주는게 좋습니다.

hwanginseop commented on Assets/Script/spawner.cs:
```diff
@@ -0,0 +1,79 @@
+            curTime = 0;
+        }
+
+        if (prefabList.Count != 0)
```
아하 고양이의 움직임을 여기서 통제하네요. 이런 방식보다는 spawner에서는 고양이의 생산만 하고. Cat.cs같이 고양이의 움직임을 제어하는 코드를 따로 제작하여 프리팹에 부착하는 방식이 스크립트의 책임면에서 더 맞는 것 같습니다.

hwanginseop commented on Assets/Script/spawner.cs:
```diff
@@ -0,0 +1,79 @@
+    {
+        GameObject clone = Instantiate(enemyObject);
+        enemyObject.transform.position = PickRandomPosition();
+        enemyObject.GetComponent<SpriteRenderer>().color = PickRandomColor();
```
이런 방식이라면 MakeRndEnemy가 호출될 때마다 캐싱하기 때문에 start나 Awake에서 한번만 캐싱 해주는게 좀 더 효율적일 것 같습니다.

hwanginseop reviewed (commented):
고생하셨습니다. 전체적으로 코드가 깔끔하네요~ 자잘한 수정사항 확인후 스스로 피드백하며 고쳐보시면 더 도움 될 것 같습니다. 추가로 변수 선언 시 접근 한정자를 적는 습관을 들이는것이 좋아보입니다. 

가람님의 코드를 예를 들어
![image](https://github.com/user-attachments/assets/cd418168-be37-47fc-816e-74085b2fb48a)
에서 rigid와 render는 접근 한정자를 적지 않았기 때문에 private으로 선언 됩니다. private인거 자체는 문제가 없지만 이를 적어놓지 않을 경우 이후 의도를 알기 힘들고 pulic 혹은 protected같은 다른 보호 수준이 필요한 경우 반드시 표기가 필요합니다.

GaMaius committed:
feat(map)

맵 타일 그려서 배경 채움.

GaMaius committed:
fix(spawner.cs), feat(enemyAI.cs)

움직임 제어 코드 분리 및 코멘트 수정사항 변경.

GaMaius committed:
feat(weapon2), fix(enemyAI.cs)

플레이어 주위를 공전하는 무기를 생성,
데미지 입히는 코드를 arrow.cs에서 enemyAI.cs로 옮겨 여러 무기에 적용 가능하게 제작함.
오류 : 플레이어 진행방향에 따라 공전하는 무기의 속도가 달라짐. 차후 개선 예정.

GaMaius committed:
Arrow 발사 수정

arrow가 가까운 적에게로 날아가게 수정됨.

GaMaius committed:
LaserGun 추가

아직 레이저 발사는 안됨.
총이 캐릭터가 바라보는 시야 안의 적 중 가장 가까운 적에게 총구를 겨눔.

GaMaius committed:
feat(Laser)

Laser구현함.
